### PR TITLE
iis logs

### PR DESCRIPTION
**Pull request by almartin82**

### CODE REVIEW

**Comment by almartin82** [comment]
Hi @Ironholds I wrote up a function to read in log data from a windows IIS server.  This PR is not yet ready for primetime - I'm seeing some funky behavior with Rcpp.

When I build and reload the package, it fails to install, throwing
```
Error: package or namespace load failed for ‘webreadr’ in dyn.load(file, DLLpath = DLLpath, ...):
 unable to load shared object '/Library/Frameworks/R.framework/Versions/3.4/Resources/library/webreadr/libs/webreadr.so':
  dlopen(/Library/Frameworks/R.framework/Versions/3.4/Resources/library/webreadr/libs/webreadr.so, 6): Symbol not found: _webreadr_internal_split_clf
  Referenced from: /Library/Frameworks/R.framework/Versions/3.4/Resources/library/webreadr/libs/webreadr.so
  Expected in: flat namespace
```

with this PR I can see what's happening pretty clearly - in `RcppExports`:

```
-    .Call(webreadr_internal_split_clf, requests)
 +    .Call(`_webreadr_internal_split_clf`, requests)
```

I'm going to throw this out to #rstats to see if anyone can help me understand, and hopefully update this PR to fix the `Rcpp` issue.

Thanks!

**Comment by almartin82** [comment]
Did some more reading on this and am realizing that I'd need to learn a lot more about Rcpp to make sense of this.  Note that my PR didn't touch the cpp functions - those were in place.  @Ironholds is it possible that this is a result of changing behavior in Rcpp?  Could it be related to https://github.com/RcppCore/Rcpp/issues/733?

**Comment by almartin82** [comment]
https://twitter.com/moneywithwings/status/936340897301704706

**Comment by Ironholds** [comment]
@almartin82 Don't worry about it - I know what the bug is! Basically it's the consequence of a (temporary) fix to an Rcpp problem that is now resolved; I'll fix it together once this is merged.

This looks like an awesome patch and contribution - one question I'd have is that the comments specify the column headings. Does that mean some columns do not have to be present, or can be in different orders?

File: R/readers.R [hunk 1]
```diff
@@ -309,4 +309,29 @@ read_s3 <- function(file){
...
+#'@examples
+#'# Using the inbuilt testing dataset
+#'iis_data <- read_iis(system.file(file.path("extdata", "iis.log"), package = "webreadr"))
+#'
+#'@export
+
+read_iis <- function(file){
+  #read in the text file to search for server restarts
+  raw_file <- readLines(file)
+  mask <- !substr(raw_file, 1, 1)=="#"
```
Comment (Ironholds) [inline R/readers.R:328]: Are there a specified number of commented lines each time? If so, a skip command might work more efficiently for large files.

File: R/readers.R [hunk 2]
```diff
@@ -309,4 +309,29 @@ read_s3 <- function(file){
...
+#'iis_data <- read_iis(system.file(file.path("extdata", "iis.log"), package = "webreadr"))
+#'
+#'@export
+
+read_iis <- function(file){
+  #read in the text file to search for server restarts
+  raw_file <- readLines(file)
+  mask <- !substr(raw_file, 1, 1)=="#"  
+  
+  names <- c("date", "time", "s-ip", "cs-method", "cs-uri-stem", "cs-uri-query", "s-port",
```
Comment (Ironholds) [inline R/readers.R:330]: I'd suggest underscores instead of dashes so it works nicely with column access.

File: R/readers.R (hunk 1 above)
Comment (almartin82) [reply R/readers.R:328]: I'm working with lots of logs and this was one of those unfortunate 'in the wild' issues that I ran into.  It's not super common, but it's very frustrating - basically, if an IIS server ever re-starts in the middle of the day, it opens up the existing log file and writes the header before it writes any data - it's at the top of every log, but it also can be in the _middle_ of any log if there was a server restart.

Comment (Ironholds) [reply R/readers.R:328]: ....good _gravy_. Okay, that makes total sense, although I might play around with a couple of ways of getting at that to test speed.

**Pull request merged**